src/main/java/org/cbio/gdcpipeline/tasklet/ProcessManifestFileTasklet.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    /*  Payload format
+     * {
+     * "filters":{
+     *          "op":"in",
+     *          "content":{
+     *                  "field":"cases.case_id",
+     *                  "value":["ABC123"]
+     *                  }
+     *          },
+     * "format":"JSON",
+     * "fields":"file_name,cases.case_id,data_format",
+     *
+     * }
+     */
+
+    protected String buildJsonRequest() {
+        JsonObject node = new JsonObject();
+        JsonObject filters = new JsonObject();
+        JsonObject content = new JsonObject();
+        JsonArray values = new JsonArray();
+        for (ManifestFileData record : manifestFileList) {
+            values.add(record.getId());
+        }
+        content.addProperty("field", "file_id");
+        content.add("value", values);
+        filters.addProperty("op", "in");
+        filters.add("content", content);
+
+        node.add("filters", filters);
+        node.addProperty("format", "JSON");
+        node.addProperty("fields", "file_id,file_name,cases.case_id,type,data_format");
+        return node.toString();
+    }
+
+    protected void gdcApiRequest(String payload) throws Exception {
+        int callCount = 0;
+        int totalCallCount = 1;
+        int from = 0;
+
+        while (callCount < totalCallCount) {
+            callCount += 1;
+            String url = GDC_API_FILES_ENDPOINT + "?from=" + from + "&size=" + MAX_RESPONSE_SIZE;
+            if (LOG.isInfoEnabled()) {
+                LOG.info(" Calling GDC API : " + url);
+                LOG.info(" Payload : " + payload);
+            }
+            GdcApiResponse res = callGdcApi(url, payload);
+            totalCallCount = res.getData().getPagination().getPages();
+            if (totalCallCount == 0) {
+                if (LOG.isErrorEnabled()) {
+                    LOG.error("Last API Request returned 0 results");
+                }
+                throw new Exception(" Last API Request returned 0 results ");
+            }
+
+            for (Hits hit : res.getData().getHits()) {
+                GdcFileMetadata g = new GdcFileMetadata();
+                g.setFile_id(hit.getFile_id());
+                g.setFile_name(hit.getFile_name());
+                g.setCases(hit.getCases());
+                g.setData_format(hit.getData_format());
+                g.setType(hit.getType());
+                gdcFileMetadatas.add(g);
+            }
+            from = callCount * MAX_RESPONSE_SIZE;
+        }
+    }
+
+    protected GdcApiResponse callGdcApi(String url, String payload) throws Exception {
+        if (manifestFileList.isEmpty()) {
+            if (LOG.isErrorEnabled()) {
+                LOG.error(" No Case Id's to add to API call");
+            }
+            throw new Exception();
+        }
+        HttpHeaders httpHeaders = new HttpHeaders();
+        httpHeaders.setContentType(MediaType.APPLICATION_JSON);
+        HttpEntity<String> entity = new HttpEntity<String>(payload.toString(), httpHeaders);
```

Comment:
You can directly parse the response into the desired java object instead of string, [here is an example](https://github.com/cBioPortal/genome-nexus-annotation-pipeline/blob/master/annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java#L133)